{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "228f246a-cadc-470e-ad78-396939a902b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: nextstrain-augur in /home/dirreno/miniconda3/lib/python3.13/site-packages (33.0.0)\n",
      "Requirement already satisfied: bcbio-gff==0.7.*,>=0.7.1 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (0.7.1)\n",
      "Requirement already satisfied: biopython==1.*,>=1.80 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (1.86)\n",
      "Requirement already satisfied: cvxopt==1.*,>=1.1.9 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (1.3.3)\n",
      "Requirement already satisfied: isodate<0.8,>=0.6 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (0.7.2)\n",
      "Requirement already satisfied: jsonschema==4.*,>=4.18.0 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (4.26.0)\n",
      "Requirement already satisfied: networkx<4,>=2.5 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (3.6.1)\n",
      "Requirement already satisfied: numpy<3,>=1 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (2.4.2)\n",
      "Requirement already satisfied: packaging>=19.2 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (25.0)\n",
      "Requirement already satisfied: pandas<3,>=1.4.0 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (2.3.3)\n",
      "Requirement already satisfied: phylo-treetime<0.12,>=0.11.2 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (0.11.4)\n",
      "Requirement already satisfied: pyfastx<3.0,>=1.0.0 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (2.3.0)\n",
      "Requirement already satisfied: python_calamine>=0.2.0 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (0.6.1)\n",
      "Requirement already satisfied: pyyaml in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (6.0.3)\n",
      "Requirement already satisfied: referencing<1.0,>=0.29.1 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (0.37.0)\n",
      "Requirement already satisfied: scipy==1.* in /home/dirreno/miniconda3/lib/python3.13/site-packages (from nextstrain-augur) (1.17.0)\n",
      "Requirement already satisfied: xopen<3,>=2.0.0 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from xopen[zstd]<3,>=2.0.0->nextstrain-augur) (2.0.2)\n",
      "Requirement already satisfied: six in /home/dirreno/miniconda3/lib/python3.13/site-packages (from bcbio-gff==0.7.*,>=0.7.1->nextstrain-augur) (1.17.0)\n",
      "Requirement already satisfied: attrs>=22.2.0 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from jsonschema==4.*,>=4.18.0->nextstrain-augur) (25.4.0)\n",
      "Requirement already satisfied: jsonschema-specifications>=2023.03.6 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from jsonschema==4.*,>=4.18.0->nextstrain-augur) (2025.9.1)\n",
      "Requirement already satisfied: rpds-py>=0.25.0 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from jsonschema==4.*,>=4.18.0->nextstrain-augur) (0.30.0)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from pandas<3,>=1.4.0->nextstrain-augur) (2.9.0.post0)\n",
      "Requirement already satisfied: pytz>=2020.1 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from pandas<3,>=1.4.0->nextstrain-augur) (2025.2)\n",
      "Requirement already satisfied: tzdata>=2022.7 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from pandas<3,>=1.4.0->nextstrain-augur) (2025.3)\n",
      "Requirement already satisfied: matplotlib>=2.0 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from phylo-treetime<0.12,>=0.11.2->nextstrain-augur) (3.10.8)\n",
      "Requirement already satisfied: isal>=1.6.1 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from xopen<3,>=2.0.0->xopen[zstd]<3,>=2.0.0->nextstrain-augur) (1.8.0)\n",
      "Requirement already satisfied: zlib-ng>=0.4.1 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from xopen<3,>=2.0.0->xopen[zstd]<3,>=2.0.0->nextstrain-augur) (1.0.0)\n",
      "Requirement already satisfied: zstandard<1 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from xopen[zstd]<3,>=2.0.0->nextstrain-augur) (0.24.0)\n",
      "Requirement already satisfied: contourpy>=1.0.1 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from matplotlib>=2.0->phylo-treetime<0.12,>=0.11.2->nextstrain-augur) (1.3.3)\n",
      "Requirement already satisfied: cycler>=0.10 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from matplotlib>=2.0->phylo-treetime<0.12,>=0.11.2->nextstrain-augur) (0.12.1)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from matplotlib>=2.0->phylo-treetime<0.12,>=0.11.2->nextstrain-augur) (4.61.1)\n",
      "Requirement already satisfied: kiwisolver>=1.3.1 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from matplotlib>=2.0->phylo-treetime<0.12,>=0.11.2->nextstrain-augur) (1.4.9)\n",
      "Requirement already satisfied: pillow>=8 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from matplotlib>=2.0->phylo-treetime<0.12,>=0.11.2->nextstrain-augur) (12.1.1)\n",
      "Requirement already satisfied: pyparsing>=3 in /home/dirreno/miniconda3/lib/python3.13/site-packages (from matplotlib>=2.0->phylo-treetime<0.12,>=0.11.2->nextstrain-augur) (3.3.2)\n"
     ]
    }
   ],
   "source": [
    "!pip install nextstrain-augur"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "60ba4540-5313-4780-95f0-afb2ae6f53cc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: augur [-h]\n",
      "             {parse,curate,merge,index,filter,subsample,mask,align,tree,refine,ancestral,translate,reconstruct-sequences,clades,traits,sequence-traits,lbi,distance,titers,frequencies,export,validate,version,import,measurements,read-file,write-file} ...\n",
      "\n",
      "Augur: A bioinformatics toolkit for phylogenetic analysis.\n",
      "\n",
      "positional arguments:\n",
      "  {parse,curate,merge,index,filter,subsample,mask,align,tree,refine,ancestral,translate,reconstruct-sequences,clades,traits,sequence-traits,lbi,distance,titers,frequencies,export,validate,version,import,measurements,read-file,write-file}\n",
      "    parse               Parse delimited fields from FASTA sequence names into\n",
      "                        a TSV and FASTA file.\n",
      "    curate              A suite of commands to help with data curation.\n",
      "    merge               Merge two or more datasets into one.\n",
      "    index               Count occurrence of bases in a set of sequences.\n",
      "    filter              Filter and subsample a sequence set. SeqKit is used\n",
      "                        behind the scenes to handle FASTA files, but this\n",
      "                        should be considered an implementation detail that may\n",
      "                        change in the future. The CLI program seqkit must be\n",
      "                        available. If it's not on PATH (or you want to use a\n",
      "                        version different from what's on PATH), set the SEQKIT\n",
      "                        environment variable to path of the desired seqkit\n",
      "                        executable. VCFtools is used behind the scenes to\n",
      "                        handle VCF files, but this should be considered an\n",
      "                        implementation detail that may change in the future.\n",
      "                        The CLI program vcftools must be available on PATH.\n",
      "    subsample           Subsample sequences from an input dataset. The input\n",
      "                        dataset can consist of a metadata file, a sequences\n",
      "                        file, or both. See documentation page for details on\n",
      "                        configuration.\n",
      "    mask                Mask specified sites from a VCF or FASTA file.\n",
      "    align               Align multiple nucleotide sequences from FASTA. The\n",
      "                        \"N\" character is treated as missing or ambiguous\n",
      "                        sites, so aligning amino acid sequences is not\n",
      "                        supported.\n",
      "    tree                Build a tree using a variety of methods. IQ-TREE\n",
      "                        specific: Strain names with spaces are modified to\n",
      "                        remove all characters after (and including) the first\n",
      "                        space.\n",
      "    refine              Refine an initial tree using sequence metadata.\n",
      "    ancestral           Infer ancestral sequences based on a tree. The\n",
      "                        ancestral sequences are inferred using `TreeTime <http\n",
      "                        s://academic.oup.com/ve/article/4/1/vex042/4794731>`_.\n",
      "                        Each internal node gets assigned a nucleotide sequence\n",
      "                        that maximizes a likelihood on the tree given its\n",
      "                        descendants and its parent node. Each node then gets\n",
      "                        assigned a list of nucleotide mutations for any\n",
      "                        position that has a mismatch between its own sequence\n",
      "                        and its parent's sequence. The node sequences and\n",
      "                        mutations are output to a node-data JSON file. If\n",
      "                        amino acid options are provided, the ancestral amino\n",
      "                        acid sequences for each requested gene are inferred\n",
      "                        with the same method as the nucleotide sequences\n",
      "                        described above. The inferred amino acid mutations\n",
      "                        will be included in the output node-data JSON file,\n",
      "                        with the format equivalent to the output of `augur\n",
      "                        translate`. The nucleotide and amino acid sequences\n",
      "                        are inferred separately in this command, which can\n",
      "                        potentially result in mismatches between the\n",
      "                        nucleotide and amino acid mutations. If you want amino\n",
      "                        acid mutations based on the inferred nucleotide\n",
      "                        sequences, please use `augur translate`. .. note:: The\n",
      "                        mutation positions in the node-data JSON are one-\n",
      "                        based.\n",
      "    translate           Translate gene regions from nucleotides to amino\n",
      "                        acids. Translates nucleotide sequences of nodes in a\n",
      "                        tree to amino acids for gene regions of the annotated\n",
      "                        features of the provided reference sequence. Each node\n",
      "                        then gets assigned a list of amino acid mutations for\n",
      "                        any position that has a mismatch between its own amino\n",
      "                        acid sequence and its parent's sequence. The reference\n",
      "                        amino acid sequences, genome annotations, and node\n",
      "                        amino acid mutations are output to a node-data JSON\n",
      "                        file. .. note:: The mutation positions in the node-\n",
      "                        data JSON are one-based.\n",
      "    reconstruct-sequences\n",
      "                        Reconstruct alignments from mutations inferred on the\n",
      "                        tree\n",
      "    clades              Assign clades to nodes in a tree based on amino-acid\n",
      "                        or nucleotide signatures. Nodes which are members of a\n",
      "                        clade are stored via <OUTPUT_NODE_DATA> → nodes →\n",
      "                        <node_name> → clade_membership and if this file is\n",
      "                        used in `augur export v2` these will automatically\n",
      "                        become a coloring. The basal nodes of each clade are\n",
      "                        also given a branch label which is stored via\n",
      "                        <OUTPUT_NODE_DATA> → branches → <node_name> → labels →\n",
      "                        clade. The keys \"clade_membership\" and \"clade\" are\n",
      "                        customisable via command line arguments.\n",
      "    traits              Infer ancestral traits based on a tree.\n",
      "    sequence-traits     Annotate sequences based on amino-acid or nucleotide\n",
      "                        signatures.\n",
      "    lbi                 Calculate LBI for a given tree and one or more sets of\n",
      "                        parameters.\n",
      "    distance            Calculate the distance between sequences across entire\n",
      "                        genes or at a predefined subset of sites.\n",
      "    titers              Annotate a tree with actual and inferred titer\n",
      "                        measurements.\n",
      "    frequencies         infer frequencies of mutations or clades\n",
      "    export              Export JSON files suitable for visualization with\n",
      "                        auspice.\n",
      "    validate            Validate files related to augur consumption or export.\n",
      "    version             Print the version of augur.\n",
      "    import              Import analyses into augur pipeline from other systems\n",
      "    measurements        Create JSON files suitable for visualization within\n",
      "                        the measurements panel of Auspice.\n",
      "    read-file           Read one or more files like Augur, with transparent\n",
      "                        optimized decompression and\n",
      "    write-file          Write a file like Augur, with transparent optimized\n",
      "                        compression and universal newlines.\n",
      "\n",
      "options:\n",
      "  -h, --help            show this help message and exit\n"
     ]
    }
   ],
   "source": [
    "!augur --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0777fe4-2a10-496b-b669-53433f3800c1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
